include/ebpf_program_attach_type_guids.h

```diff
@@ -81,6 +81,13 @@ extern "C"
     __declspec(selectany) ebpf_attach_type_t EBPF_ATTACH_TYPE_SAMPLE = {
         0xf788ef4b, 0x207d, 0x4dc3, {0x85, 0xcf, 0x0f, 0x2e, 0xa1, 0x07, 0x21, 0x3c}};
 
+    /** @brief Attach type for handling incoming packets as early as possible.
+     *
+     * Program type: \ref EBPF_PROGRAM_TYPE_XDP_TEST
+     */
+    __declspec(selectany) ebpf_attach_type_t EBPF_ATTACH_TYPE_XDP_TEST = {
```

Comment:
Along the lines of #2305, test/sample hooks should not have definitions in core eBPF headers.

@saxena-anurag has #3055 in progress, which should provide a framework within this repo to separate test/sample code from the core modules. I'm not sure if the two PRs should be ordered relative to each other?

Reply:
good point - we'll make sure to stage things appropriately (we'll need one change to go in first, and then have the other rebase after completion).